tooling/minitest/src/tests/builder_api.rs

```diff
@@ -0,0 +1,115 @@
+use crate::*;
+
+#[test]
+fn global_var() {
+    let mut p = ProgramBuilder::new();
+    let var = p.global_int::<u32>();
+    let mut f = p.function();
+    f.assign(var, const_int(42u32));
+    f.if_(eq(load(var), const_int(42u32)), |_| {}, |f| f.unreachable());
+    f.exit();
+    p.finish_function(f);
+    let p = p.finish_program();
+    assert_stop(p);
+}
+
+#[test]
+fn local_var() {
+    let mut p = ProgramBuilder::new();
+    let mut f = p.function();
+    let var = f.local::<u32>();
+    f.storage_live(var);
+    f.assign(var, const_int(42u32));
+    f.if_(eq(load(var), const_int(42u32)), |_| {}, |f| f.unreachable());
+    f.exit();
+    p.finish_function(f);
+    let p = p.finish_program();
+    assert_stop(p);
+}
+
+#[test]
+fn arg_and_ret_var() {
+    let mut p = ProgramBuilder::new();
+    let mut f = p.function();
+    let var = f.arg::<u32>();
+    let ret = f.ret::<u32>();
+    f.if_(eq(load(var), const_int(42u32)), |_| {}, |f| f.unreachable());
+    f.assign(ret, add(load(var), const_int(2u32)));
+    f.exit();
+    let add_two_if_42 = p.finish_function(f);
+
+    let mut start = p.function();
+
+    let l = start.local::<u32>();
+    start.storage_live(l);
+    start.call(add_two_if_42, &[by_value(const_int(42u32))], l, Ret::Yes);
```

Comment:
I think the order of arguments for `call` should be like in `x = fun(arg1, arg2)`, i.e. with the return place first.

Also, we shouldn't have the `Ret` argument, it's annoying to repeat that all the time. Instead we can have a `call_noret` function for the rare case where a function we call cannot return.